### ISSUE TITLE
Subscriber role shows nav links to screens the role doesn't have access to

### ISSUE THREAD

**Issue by ash1eygrace** [Automattic/wp-calypso#62975]
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Invite a WordPress.com User as a Subscriber on an Atomic or Jetpack connected site via My Site > Users invite. 
2. Log in to the Subscriber's WordPress.com account > Click on My Site > Choose the site you're subscribed to. 
3. Click on the links you shouldn't have access to as a subscriber and get various permission on-page errors, banners, or Console errors.

### A clear and concise description of what you expected to happen.

I'd expect to see a link to view the site's front end since I'm only a subscriber. 

At the least, I'd expect not to see the links that my role can't access, as they just load broken pages due to lack of permission. 

### What actually happened

I see the following links in the dashboard of the site I'm subscribed to:

- Stats
- Upgrades
- Posts
- Media
- Pages
- Comments
- Feedback
- Jetpack
- Appearance
- Plugins
- Users
- Tools
- Settings
- WP Admin

When I click on any of the Nav links above other than published Pages or Posts, I get permission errors, which is expected since subscribers can only view posts, and pages, and leave comments if enabled. Technically, the role-based permissions are working (I can't access things I don't have permission to as a subscriber), but it looks really broken and alarming when every page either won't load or tells me I don't have permission to view the page: 

https://user-images.githubusercontent.com/29527450/163636153-b75c791a-3e21-4694-a03b-b3860ba6f67b.mp4

 

### Operating System

macOS

### Browser

Chrome/Chromium

**Comment by coder-karen** [comment]
Yes I can also see the same - only if inviting a user as a subscriber though via the WordPress.com interface with the steps you gave (but not adding them as a subscriber to the site itself in wp-admin). Thanks for the great screencast showing this. It would make sense to hide some of these menu items.

**Comment by jeherve** [comment]
Nice catch!

Since that interface (the WordPress.com dashboard) is developed in the Calypso repo, I'll transfer the issue there.

**Comment by mxhassani** [comment]
Similar issue here: 5166577-zen
Was able to replicate on test site

**Comment by cuemarie** [comment]
📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - Yes

📌 SCRUBBING
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 

📌 FINDINGS/SCREENSHOTS/VIDEO
- Replicable just as Ashley describes - once an invited subscriber clicks `Accept Invitation` in their email, they are taken to a WordPress.com screen where they are prompted to `Join` the site, and then they land on a dashboard of the site - already pretty confusing, if you're just trying to view pages posts and comments. Further, much of the dashboard is inaccessible.

Recording of a WordPress.com user's experience being added as a subscriber to a Self-Hosted WordPress site:

https://user-images.githubusercontent.com/27249804/172451080-316f0bd5-938f-44a5-928c-93ec14337890.mov

📌 ACTIONS
- Marked as Triaged for Quality Squad review
- Added to 'HE Cross-repo Watchlist' project board

- I've asked @Automattic/dotcom-pqa-group to weigh in on whether this needs to transfer anywhere else

**Comment by happychait** [comment]
- [ ] Another report in 5354848-zen

Subscriber invited to this site is actually able to see the nav links that are meant for the Contributor role, in combo with https://github.com/Automattic/wp-calypso/issues/63978
